src/main/sensors/battery.c

```diff
@@ -159,8 +159,21 @@ void batteryUpdateVoltage(timeUs_t currentTimeUs)
             break;
     }
 
+    if (!voltageMeter.isVoltageStable && cmp32(millis(), voltageMeter.prevDisplayFilteredTime) >= PREV_DISPLAY_FILTERED_TIME_DIFF) {
+        if ((voltageState == BATTERY_NOT_PRESENT || voltageState == BATTERY_INIT) && isVoltageFromBattery()) {
```

Comment:
Due to `isVoltageFromBattery()`, `isVoltageStable` is never set unless battery is connected.